### PR TITLE
Policy Tag: Add IAM at tag level

### PR DESCRIPTION
**Pull request by lcaggio**
Data Catalog policy tag: Add IAM at tag level

### CODE REVIEW

File: modules/data-catalog-policy-tag/README.md [hunk 1]
```diff
@@ -31,6 +31,23 @@ module "cmn-dc" {
...
 ```
+
+### Taxonomy with Tag level IAM binding
+
+```hcl
+module "cmn-dc" {
+  source     = "./modules/data-catalog-policy-tag"
+  name       = "my-datacatalog-policy-tags"
+  project_id = "my-project"
+  tags       = ["low", "medium", "high"]
```
Comment (ludoo) [inline modules/data-catalog-policy-tag/README.md:42]: Lorenzo, I would prefer having a single variable

```hcl
tags = {
  foo = null
  bar = {
    "roles/foobar" = ["user:a@b.com"]
  }
}
```

**Review by ludoo — CHANGES_REQUESTED** [review]

**Comment by ludoo** [comment]
Can you add a test that check the different values for the new variable? null etc

File: modules/data-catalog-policy-tag/variables.tf [hunk 2]
```diff
@@ -72,7 +72,7 @@ variable "project_id" {
 }
 
 variable "tags" {
-  description = "List of Data Catalog Policy tags to be created."
-  type        = list(string)
-  default     = []
+  description = "List of Data Catalog Policy tags to be created with optional IAM binging configuration in {tag => {ROLE => [MEMBERS]}} format."
+  type        = map(map(list(string)))
+  default     = {}
```
Comment (ludoo) [inline modules/data-catalog-policy-tag/variables.tf:77]: This should also have `nullable = false`.

**Review by ludoo — APPROVED** [review]

**Comment by lcaggio** [comment]
Policy Tag: Add IAM at tag level

**Comment by lcaggio** [comment]
Policy Tag: Add IAM at tag level.

**Comment by lcaggio** [comment]
Policy Tag: Add IAM at tag level.
